src/test/java/net/masterthought/cucumber/json/ElementTest.java

```diff
@@ -0,0 +1,120 @@
+package net.masterthought.cucumber.json;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import net.masterthought.cucumber.Configuration;
+import net.masterthought.cucumber.FileReaderUtil;
+import net.masterthought.cucumber.json.support.Status;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class ElementTest {
+    // a simple element json that we'll tweak for our tests
+    private static final String TEMPLATE_JSON = "json/element/template.json";
+
+    @Parameterized.Parameters
+    public static Collection<Status[]> data() {
+        Collection<Status[]> params = new ArrayList<>();
+        params.add(new Status[] {Status.PASSED});
+        params.add(new Status[] {Status.FAILED});
+        params.add(new Status[] {Status.SKIPPED});
+        params.add(new Status[] {Status.MISSING});
+        params.add(new Status[] {Status.PENDING});
+        params.add(new Status[] {Status.UNDEFINED});
+        return params;
+    }
+
+    // the status we're testing for this run
+    private final Status step_status;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/damianszczepanik/cucumber-reporting/file/2497471963/issues/source?bid=3059308&fileBranchId=3270310#l39)